src/Scanner.js

```diff
@@ -143,19 +149,19 @@ class DefaultScanner {
             return this;
         }
 
-        if ('matches' in context && context.matches?.(this.matching.cssSelector('formInputsSelectorWithoutSelect'))) {
+        const formInputsSelectorWithoutSelect = this.matching.cssSelector('formInputsSelectorWithoutSelect');
+
+        if ('matches' in context && context.matches?.(formInputsSelectorWithoutSelect)) {
             this.addInput(context);
         } else {
-            const selector = this.matching.cssSelector('formInputsSelectorWithoutSelect');
-            const inputs = context.querySelectorAll(selector);
+            const inputs = context.querySelectorAll(formInputsSelectorWithoutSelect);
             if (inputs.length > this.options.maxInputsPerPage) {
                 this.setMode('stopped', `Too many input fields in the given context (${inputs.length}), stop scanning`, context);
                 return this;
             }
             inputs.forEach((input) => this.addInput(input));
             if (context instanceof HTMLFormElement && this.forms.get(context)?.hasShadowTree) {
-                const selector = this.matching.cssSelector('formInputsSelectorWithoutSelect');
-                findEnclosedElements(context, selector).forEach((input) => {
+                findElementsInShadowTree(context, formInputsSelectorWithoutSelect).forEach((input) => {
```

Comment:
No-op refactor, just reusing the variable.

Reply:
Nice!